### ISSUE TITLE
Replace uses of os.path with pathlib in settings.py

### ISSUE THREAD

**Issue by kimberscott** [lookit/lookit-api#569]
**TL;DR**: Use pathlib and Path objects rather than strings constructed by os.path.join in settings.

**Narrative**
As a Developer, I want to use pathlib for path manipulation as it's more straightforward to use and will make it easier to ensure paths work correctly for local development independent of platform.

**Acceptance Criteria**
- [ ] No use of os.path.join in settings.py, paths constructed via pathlib
- [ ] Existing tests pass

**Implementation Notes**
See discussion here: https://github.com/lookit/lookit-api/pull/552#discussion_r483068067

We didn't switch immediately because Path objects couldn't be used for settings; we'll need to update Django to 3.1 for that.
